client/blocks/domain-to-plan-nudge/index.jsx

```diff
@@ -0,0 +1,333 @@
+		}
+		const noticeAction = this.props.infoNotice( translate( 'Submitting payment' ) );
+		this.setState( {
+			isSubmitting: true,
+			noticeId: get( noticeAction, 'notice.noticeId' )
+		} );
+
+		submitTransaction( { cart, transaction }, this.handleTransactionComplete );
+	}
+
+	render() {
+		if ( ! this.isVisible() ) {
+			return null;
+		}
+
+		const { isSubmitting } = this.state;
+
+		const {
+			discountedRawPrice,
+			productSlug,
+			rawDiscount,
+			rawPrice,
+			siteId,
+			storedCard,
+			translate,
+			userCurrency
+		} = this.props;
+
+		return (
+			<DismissibleCard
+				className="domain-to-plan-nudge"
+				preferenceName="domain-to-plan-nudge"
+				temporary
+			>
+				<QueryStoredCards />
+				{ siteId && <QuerySitePlans siteId={ siteId } /> }
+
+				<div className="domain-to-plan-nudge__header">
+					<div className="domain-to-plan-nudge__header-icon">
+						<PlanIcon plan={ productSlug } />
+					</div>
+					<div className="domain-to-plan-nudge__header-copy">
+						<h3 className="domain-to-plan-nudge__header-title">
+							{ translate( 'Upgrade to a Personal Plan and Save!' ) }
+						</h3>
+						<ul className="domain-to-plan-nudge__header-features">
+							<li>
+								<div className="domain-to-plan-nudge__header-features-item">
+									<Gridicon
+										className="domain-to-plan-nudge__header-features-item-checkmark"
+										icon="checkmark"
+										size={ 24 }
+									/>
+									{ translate( 'Remove all WordPress.com advertising from your website' ) }
+								</div>
+							</li>
+							<li>
+								<div className="domain-to-plan-nudge__header-features-item">
+									<Gridicon
+										className="domain-to-plan-nudge__header-features-item-checkmark"
+										icon="checkmark"
+										size={ 24 }
+									/>
+									{ translate( 'Get high quality live chat and priority email support' ) }
+								</div>
+							</li>
+							<li>
+								<div className="domain-to-plan-nudge__header-features-item">
+									<Gridicon
+										className="domain-to-plan-nudge__header-features-item-checkmark"
+										icon="checkmark"
+										size={ 24 }
+									/>
+									{ translate( 'Upload up to 3GB of photos and documents' ) }
+								</div>
+							</li>
+							<li>
+								<div className="domain-to-plan-nudge__header-features-item">
+									<Gridicon
+										className="domain-to-plan-nudge__header-features-item-checkmark"
+										icon="checkmark"
+										size={ 24 }
+									/>
+									{ translate( 'Bundled with your domain for the best value!' ) }
+								</div>
+							</li>
+						</ul>
+					</div>
+				</div>
+				<div className="domain-to-plan-nudge__actions-group">
+					<div className="domain-to-plan-nudge__plan-price-group">
+						<div
+							className="domain-to-plan-nudge__discount-value">
+							{
+								translate( 'SAVE %(discount)s', {
+									args: {
+										discount: formatCurrency( rawDiscount, userCurrency )
```

Comment:
I considered not displaying elements, but wanted to show what this looks like all together. We can add a discount visibility check later in `isVisible` to avoid rendering this at all in this case.